### DISCUSSION TITLE
What is the main difference to https://github.com/locuslab/differentiable-mpc

### DISCUSSION THREAD

**Discussion by roangel** [facebookresearch/theseus#273] [Q&A] [answered]
Hi guys,

I was wondering, what are the main differences between Theseus and https://github.com/locuslab/differentiable-mpc?

Thanks!

**Comment by bamos** [comment, marked as answer]
Theseus here is for solving and differentiating through general non-linear least squares optimization problems while that differentiable MPC focuses on solving control problems with iLQR and box DDP. While we have some motion planning problems here in Theseus, they're not the same as the control problems solved with MPC. To start going deeper into these differences, I recommend reading and comparing the [DMPC paper](https://arxiv.org/abs/1810.13400) with the [DGPMP paper](https://arxiv.org/abs/1907.09591), and then going through everything else we have here in the [Theseus paper](https://arxiv.org/abs/2207.09442).

**Reply by roangel** [reply]
Thanks a lot! :)
